ruff.toml

```diff
@@ -14,6 +14,7 @@ ignore = [
     "ISC001",
     "TCH002",
     "PLW1514", # allow Path.open without encoding
+    "S603"# https://github.com/astral-sh/ruff/issues/4045
```

Comment:
Oops, I misread the comments and mixed up the rules. That said, S603 is a very valid rule so it shouldn't be blanket-disabled; only locally in scripts as necessary.

Reply:
+1

Reply:
Hmm, after further reading and testing (including with the latest `ruff`, `0.6.8`, rather than the fairly old version pinned in the tox config, it does seem that it’s practically impossible to use `subprocess` without _one_ of the S### rules being triggered, so I think this can stay in the global exclusions. That said, I will earmark a run through of them myself in the near future as `PLW1514` seems like one that should not be excluded, for exampe.

Reply:
I spent a few iterations bouncing between S603 and S602 on that `subprocess` call.  Based on the info in that link from the `ruff` team, it seems appropriate to leave S603 as a global exclusion, otherwise the next introduction of `subprocess` will encounter the same issue.

Reply:
yeah, the rule is a bit much, but hard to say we won't need `subprocess`. not for this PR, but is there a way to make our own rules?

Reply:
ruff does not yet support custom rules: https://github.com/astral-sh/ruff/issues/283